File: Easy/121-best-time-to-buy-and-sell-stock/try-14.cpp
/*
 * Problem #121: Best Time to Buy and Sell Stock
 * Difficulty: Easy
 * Submission: Try 14
 * status: Accepted
 * Language: cpp
 * Date: 04/02/2025, 23:59:07
 * Link: https://leetcode.com/problems/best-time-to-buy-and-sell-stock/
 */

class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int maxi = 0,mini = prices[0];
        for(auto i : prices){
            maxi = max(maxi,i - mini);
            mini = min(mini,i);
        }
        return maxi;
    }
};

File: Easy/387-first-unique-character-in-a-string/try-14.py
/*
 * Problem #387: First Unique Character in a String
 * Difficulty: Easy
 * Submission: Try 14
 * status: Accepted
 * Language: python3
 * Date: 07/12/2024, 09:16:11
 * Link: https://leetcode.com/problems/first-unique-character-in-a-string/
 */

class Solution:
    def firstUniqChar(self, s: str) -> int:
        c=Counter(s)
        for i in range(len(s)):
            if c[s[i]]==1:
                return i
        return -1
